flink-java/src/main/java/org/apache/flink/api/java/functions/SelectByMinFunction.java

```diff
@@ -72,8 +73,8 @@ public T reduce(T value1, T value2) throws Exception {
 		for (int position : fields) {
 			// Save position of compared key
 			// Get both values - both implement comparable
-			Comparable comparable1 = value1.getFieldNotNull(position);
-			Comparable comparable2 = value2.getFieldNotNull(position);
+			Comparable comparable1 = ((Tuple)value1).getFieldNotNull(position);
```

Comment:
I have a doubt here. Previously also this SelectByMaxFunction was extending T extends Tuple. So TupleTypeInfo was actually allowed here. So even now we are allowing TypeInformation<T> which is the super class of TupleTypeInfo. So it should work the same way as it was earlier?